### ISSUE TITLE
Any plans to release this for Linux?

### ISSUE THREAD

**Issue by dbruning** [microsoft/MixedReality-WebRTC#174]
The readme says: 

> ... planned support for Unity deployment on iOS and Android

Are there any plans to support deployment (without Unity) onto linux, e.g. ubuntu? This would be a great way to stream content out from embedded devices like the nvidia jetson.

**Comment by djee-ms** [comment]
No there is no plan at the moment. We have a [long list](#142) of most-awaited features to implement for the next 2.0 release, and we chose to scope things to Unity-only support to help us deal with the amount of work without delaying further support for Android. iOS support is tentatively scheduled for the release after that, again because of limited dev resources, and based on the relative lower signal we got from partners and the community compared with Android. As always though for everything, if we see some strong signal that support for native Android and/or Linux is of interest then we will consider scheduling that feature too. But this has not really come up so far.

**Comment by dbruning** [comment]
Thanks for the response, I understand that you'll have priorities. I'll tell you a bit more about my use-case for the record anyway.

You'll obviously be aware that Microsoft have released the Kinect for Azure camera (essentially a repackaging of the Hololens camera). [Support for Linux](https://feedback.azure.com/forums/920053-azure-kinect-dk/suggestions/37615903-arm-support-for-linux) is the second-highest [uservoice](https://feedback.azure.com/forums/920053-azure-kinect-dk/suggestions/37615903-arm-support-for-linux) item for the SDK, and is [planned for the end of February](https://github.com/microsoft/Azure-Kinect-Sensor-SDK/pull/870#issuecomment-555230934). 

Imagine the following architecture:

- a Blazor application running on nvidia jetson nano (ubuntu)
- the Blazor application takes depth and color information from the Kinect Azure camera via the SDK, and processes it using the Jetson's GPU
- the Blazor application provides a UI from the application that any phone/tablet/computer on the local network can connect to
- if we had a good way to use WebRTC, the Blazor application could also easily share out a "livestream" from the camera as part of the application UI.

All the pieces are in place for that now, apart from a good WebRTC c# wrapper that works on Linux.

**Comment by djee-ms** [comment]
Thanks for the feedback and use case description, this is always helpful for us to prioritize.

About Azure Kinect DK, I think your use case works around that or is not affected, but:
- 7-channel audio (microphone array) is not supported currently by WebRTC, and there is no plan from Google to do so, as there seem to be many parts of the audio engine not designed for anything else than mono/stereo. See discussion with the Google WebRTC dev team [here](https://bugs.chromium.org/p/webrtc/issues/detail?id=10881).
- Multi-channel video is not supported. You can work around by using multiple video tracks, but I am not sure whether frames stay in sync with each other in this case, and depending on the application that might be an issue (_e.g._ color frame N arrives at the same time as depth frame N+1). In any case I don't think this was ever tested.

**Comment by dbruning** [comment]
Yep I understand those limitations - in my use case the camera frames are processed on the Jetson Nano, and just the result of that processing (a marked-up video feed, can be relatively low-resolution, plus some limited numeric data) needs to be passed out via WebRTC.
